core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java

```diff
@@ -24,15 +26,34 @@ public class LogErrorHandler implements Handler<RoutingContext> {
 
   @Override
   public void handle(final RoutingContext failureContext) {
-
     if (failureContext.failed()) {
-      LOG.error(
-          String.format("Failed request: %s", failureContext.request().absoluteURI()),
-          failureContext.failure());
+      final String requestUri = getRequestUri(failureContext);
+
+      if (failureContext.failure() != null) {
+        LOG.error(
+            "Failed request: {} due to {}",
+            requestUri,
+            failureContext.failure().getMessage(),
+            failureContext.failure());
+      } else {
+        LOG.error("Failed request: {}", requestUri);
+      }
+
       // Let the next matching route or error handler deal with the error, we only handle logging
       failureContext.next();
     } else {
       LOG.warn("Error handler triggered without any propagated failure");
     }
   }
+
+  private static String getRequestUri(RoutingContext failureContext) {
```

Comment:
I think the method name isn't quite right as the error case it doesn't return a requestUri. Would be better to return an empty and just include that error message in the handle method.

Reply:
agreed. Updated the method to return Optional.